### PR TITLE
Update README instructions

### LINKED ISSUE #884: Update README instructions
Following the guide I ran into a number of issues which I feel we can address by updating the documentation:

1. Suggest Mac, there are a number of complications when trying to set up on Windows.
2. Remove deploying to website section, this is not purely for external use and can be moved into a confluence page.
3. Mention the need for Ruby 2.7.0, this is needed to run * `bundle install` to install the necessary gems, including `nanoc`
4. Remove Docker instructions while we work on fixing https://ably.atlassian.net/browse/WEB-383

TBC

### PR DESCRIPTION
**Pull request by MarkWoulfeAbly**
Fixes #884

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,32 +1,52 @@
 # Ably Documentation Overview
 
-[Ably](https://www.ably.io) is a hugely scalable, superfast and secure hosted realtime messaging service for web-enabled devices.
+[Ably](https://www.ably.io) is a scalable, fast, and secure hosted realtime messaging service for web-enabled devices.
 
-This Git repository contains most of the [Ably API documentation](https://www.ably.io/documentation) that resides at <https://www.ably.io/documentation> in [Textile](redcloth.org/textile) format.  [Nanoc](http://nanoc.stoneship.org/) is used to build the static site from the `master` branch.
+The static site generated from this repository is hosted at <http://docs.ably.io> and deployed automatically when the `master` branch is updated.
```
Comment (tomczoink) [inline README.md:5]: Extra space on end of line.

File: README.md [hunk 2]
```diff
@@ -1,32 +1,52 @@
 # Ably Documentation Overview
 
-[Ably](https://www.ably.io) is a hugely scalable, superfast and secure hosted realtime messaging service for web-enabled devices.
+[Ably](https://www.ably.io) is a scalable, fast, and secure hosted realtime messaging service for web-enabled devices.
 
-This Git repository contains most of the [Ably API documentation](https://www.ably.io/documentation) that resides at <https://www.ably.io/documentation> in [Textile](redcloth.org/textile) format.  [Nanoc](http://nanoc.stoneship.org/) is used to build the static site from the `master` branch.
+The static site generated from this repository is hosted at <http://docs.ably.io> and deployed automatically when the `master` branch is updated.  
 
-The Ably documentation was intentionally created as a public repository using a simple text based markup language so that:
+We frequently publish updates from this repository so is typically more up to date than the official Ably documentation which can be found at <https://www.ably.io/documentation>
```
Comment (tomczoink) [inline README.md:7]: Missing full stop at end of line.

File: README.md [hunk 3]
```diff
@@ -1,32 +1,52 @@
...
 
-* It can be maintained by Ably staff and developers.
-* 3rd parties can fork the repository, and submit pull requests to improve the documentation.  We welcome the community's suggestions and input.
-* This documentation is later merged upstream to the [Ably primary website documentation](https://www.ably.io/documentation) with each website release.
+This Git repository uses the [Textile](redcloth.org/textile) format with [Nanoc](http://nanoc.stoneship.org/) used to build the static site from the `master` branch.
```
Comment (tomczoink) [inline README.md:9]: redcloth.org/textile missing https:// (currently directs to the https://github.com/ably/docs/blob/a41414363053940af89c985e4846d781fbae3b9f/redcloth.org/textile).

File: README.md [hunk 4]
```diff
@@ -1,32 +1,52 @@
...
 
-## Viewing the documentation
+## Running locally
 
-The static site generated from this documentation repository is hosted at <http://docs.ably.io> and deployed automatically when the `master` branch is updated.  We frequently publish updates from this repository so is typically more up to date than the [official Ably API documentation](https://www.ably.io/documentation).
+1. Clone a local version of the repo, or a local version of your forked repo.
+3. Checkout the master branch.
```
Comment (tomczoink) [inline README.md:14]: Numbering goes 1-3-4 (Markdown makes it look correct when rendered, but probably better to either make the numbering match correctly, or go for the old style 1-1-1 for consistency).

File: README.md [hunk 5]
```diff
@@ -49,48 +69,6 @@ which gets published
 The code that is published to JSBin can be found in
 [content/code/](content/code/).
 
-## Forking and running locally
```
Comment (tomczoink) [inline README.md:52]: The link above still tries to point to this:

**This configuration is not required for basic documentation creation and modification, so most
editors can simply skip to
[Forking and running locally](#forking-and-running-locally).**

File: README.md (hunk 1 above)
Comment (MarkWoulfeAbly) [reply README.md:5]:
```suggestion
The static site generated from this repository is hosted at <http://docs.ably.io> and deployed automatically when the `master` branch is updated.
```

File: README.md (hunk 2 above)
Comment (MarkWoulfeAbly) [reply README.md:7]:
```suggestion
We frequently publish updates from this repository so is typically more up to date than the official Ably documentation which can be found at <https://www.ably.io/documentation>.
```

File: README.md (hunk 3 above)
Comment (MarkWoulfeAbly) [reply README.md:9]:
```suggestion
This Git repository uses the [Textile](https://redcloth.org/textile) format with [Nanoc](http://nanoc.stoneship.org/) used to build the static site from the `master` branch.
```

File: README.md [hunk 6]
```diff
@@ -1,32 +1,52 @@
...
-* This documentation is later merged upstream to the [Ably primary website documentation](https://www.ably.io/documentation) with each website release.
+This Git repository uses the [Textile](https://redcloth.org/textile) format with [Nanoc](http://nanoc.stoneship.org/) used to build the static site from the `master` branch.
 
-## Viewing the documentation
+## Running locally
 
-The static site generated from this documentation repository is hosted at <http://docs.ably.io> and deployed automatically when the `master` branch is updated.  We frequently publish updates from this repository so is typically more up to date than the [official Ably API documentation](https://www.ably.io/documentation).
+1. Clone a local version of the repo, or a local version of your forked repo.
+3. Checkout the master branch.
+4. Create and checkout your feature or fix branch.
```
Comment (MarkWoulfeAbly) [inline README.md:15]:
```suggestion
1. Clone a local version of the repo, or a local version of your forked repo.
2. Checkout the master branch.
3. Create and checkout your feature or fix branch.
```

**Comment by MarkWoulfeAbly** [comment]
Thanks @tomczoink. @mattheworiordan as most of this was written and structured by you let me know if there is anything you fundamentally disagree with in these changes.

File: README.md [hunk 7]
```diff
@@ -1,95 +1,69 @@
...
-If you will be creating or modifying code within this repository that is to be uploaded automatically
-to [JSBin](https://jsbin.ably.io/), for our "Try it now code editor",
-then you will need to create yourself a JSBin config file:
+_Note: On Windows you will need to re-run the `docker-compose up` command to see changes, unless you are using [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10)._
 
-    cp config/jsbin_config.example.yaml config/jsbin_config.yaml
+### Running with Ruby
 
-And then you will need to populate the `api_key` property in that file with a JSBin API key.
+Using Ruby directly (2.7.0) you must first run the following command:
```
Comment (mattheworiordan) [inline README.md:33]: I would avoid specifying the Ruby version here, and instead link to the `.ruby_version` file stating that version should be used

File: README.md [hunk 8]
```diff
@@ -1,95 +1,69 @@
...
+The static site generated from this repository is hosted at <http://docs.ably.io> and deployed automatically when the `master` branch is updated.
 
-The Ably documentation was intentionally created as a public repository using a simple text based markup language so that:
+We frequently publish updates from this repository so is typically more up to date than the official Ably documentation which can be found at <https://www.ably.io/documentation>.
 
-* It can be maintained by Ably staff and developers.
-* 3rd parties can fork the repository, and submit pull requests to improve the documentation.  We welcome the community's suggestions and input.
-* This documentation is later merged upstream to the [Ably primary website documentation](https://www.ably.io/documentation) with each website release.
+This Git repository uses the [Textile](https://redcloth.org/textile) format with [Nanoc](http://nanoc.stoneship.org/) used to build the static site from the `master` branch.
```
Comment (mattheworiordan) [inline README.md:9]: I think it's important we include a reference to the documentation on how to write content in `content/client-lib-development-guide/documentation-formatting-guide.textile` as anyone who wants to contribute, would need to read that to get going.

File: README.md [hunk 9]
```diff
@@ -1,95 +1,69 @@
...
+2. Commit your changes, push your branch and send us a pull request.
+3. All pull requests will automatically launch a static review site that will be linked to at the bottom of the PR.
 
-_This has been tested with Docker version 2.3.0.3. See [Docker installation instructions](https://docs.docker.com/get-docker/)._
-
-## Running on Heroku
+## Publishing to JSBin
 
-This repo will automatically run on Heroku, but relies on the following buildpacks (see https://github.com/ably-forks/heroku-buildpack-nanoc):
+If you will be creating or modifying code within this repository that is to be uploaded automatically
```
Comment (mattheworiordan) [inline README.md:53]: I think this is a bit ambiguous.  There is code all over the place in our repository.  JSBin only applies to code in the code folder.  I also think we should link to the JSBin repo with information on how that works as there are docs in there too.

File: README.md [hunk 10]
```diff
@@ -113,15 +87,37 @@ with that major/minor version.
...
 fix to be incorporated until the next spec release.
 
-When updating a client lib to a spec version, client lib devs should
+When updating a client lib to a spec version, client lib developers should
 work from a diff from the tag of the previous release, so as to
 incorporate all changes since that tag.
 
+## Deploying to website (www.ably.io/documentation)
+
+The website consumes this repo through a ruby gem. The gem points at the `master` branch and saves the revision as a version. To release to `/documentation`, follow these steps:
```
Comment (mattheworiordan) [inline README.md:101]: Minor point - ruby is referred to as Ruby elsewhere on this page

Comment (MarkWoulfeAbly) [reply README.md:101]:
```suggestion
The website consumes this repo through a Ruby gem. The gem points at the `master` branch and saves the revision as a version. To release to `/documentation`, follow these steps:
```

**Pull request merged**